lib/sda/tree.go

```diff
@@ -0,0 +1,389 @@
+// topology is a general
+package sda
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/edwards"
+	"github.com/satori/go.uuid"
+)
+
+// In this file we define the main structures used for a running protocol
+// instance. First there is the Entity struct: it represents the Entity of
+// someone, a server over the internet, mainly tied by its public key.
+// The tree contains the peerId which is the ID given to a an Entity / server
+// during one protocol instance. A server can have many peerId in one tree.
+// ProtocolInstance needs to know:
+// - which EntityList we are using ( a selection of proper servers )
+// - which Tree we are using.
+// - The overlay network: a mapping from PeerId
+// It contains the PeerId of the parent and the sub tree of the children.
+
+// Tree is a topology to be used by any network layer/host layer
+// It contains the peer list we use, and the tree we use
+type Tree struct {
+	Id         uuid.UUID
+	EntityList *EntityList
+	Root       *TreeNode
+}
+
+var TreeType = network.RegisterMessageType(Tree{})
+
+// NewTree creates a new tree using the entityList and the root-node. It
+// also generates the id.
+func NewTree(il *EntityList, r *TreeNode) *Tree {
+	r.UpdateIds()
+	url := "https://dedis.epfl.ch/tree/" + il.Id.String() + r.Id.String()
```

Comment:
A global config variable is also fine I guess. I just think it should be put somewhere where it's more visible in case other people want to run their own cothority network.